calc/l3_route_resolver.go

```diff
@@ -763,11 +789,12 @@ type RouteInfo struct {
 		NodeNames []string // set if this CIDR _is_ a node's own IP.
 	}
 
-	// LocalWEP contains information extracted from the local workload endpoints.
-	LocalWEP struct {
-		// Count of local WEPs that have this CIDR.  Normally, this will be 0 or 1 but Felix has to be tolerant
-		// to bad data (two local WEPs with the same CIDR) so we do ref counting.
-		RefCount int
+	// WEP contains information extracted from the workload endpoints.
+	WEP struct {
+		// Count of WEPs that have this CIDR.  Normally, this will be 0 or 1 but Felix has to be tolerant
+		// to bad data (two WEPs with the same CIDR) so we do ref counting.
+		// The RefCount is tracked per-node, to properly handle remote weps.
+		RefCount map[string]int
```

Comment:
I made this a map since previously the RefCount could only apply to local WEPs and so the node name was implicitly known.

We now need to track the node name for each WEP, and we don't want to blat over it if we end up in a transient scenario where two CIDRs are owned by different WEPs on different nodes, and thus end up with a route pointing at the wrong node. This seems to let us maintain both nodenames, and will eventually resolve itself down to a single entry.

LMK if you have better ideas on how to do this!

Reply:
I think you're right that we need to solve that problem.  I might be tempted to optimise for occupancy rather than efficiency.  For example, you could store the same information as a `[]struct{Node string, RefCount int}`; in the mainline case where there's exactly 1, it'd probably be faster than a map anyway and you could sort the slice on nodename on update so that lookup is O(1) (pick the 0th entry)